File: Programmers/모음사전.py
from itertools import product
def solution(word):
    answer = 0
    vowels = ['A','E','I','O','U']
    all_words = []
    words_sequence = {}
    for i in range(1,6):
        all_words.extend(list(product(vowels, repeat=i)))
    
    all_words.sort()
    for seq, word_value in enumerate(all_words):
        if "".join(word_value) == word:
            return seq + 1

File: Programmers/모음사전.kt
class Solution {
    
    var cnt = 0
    var flag = false
    
    fun solution(word: String): Int {
        var answer = 0
        var vowels = arrayOf('A','E','I','O','U')
        dfs("",vowels, word)
    
        return cnt
    }
    
    fun dfs(w:String, vowels:Array<Char>, word:String){
        if (w.length > 5){
            return
        }
        else{
            if (w.length >0){
                cnt += 1
            }
            if (w.equals(word)){
                flag = true
            }

            for(i in 0..4){
                if(flag == false)
                    dfs(w+vowels[i],vowels,word)
            }
        }
    }


}